streams/test-utils/src/test/java/org/apache/kafka/streams/TopologyTestDriverTest.java

```diff
@@ -897,4 +900,120 @@ public void close() {}
             );
         }
     }
+    
+    @Test
+    public void shouldFeedStoreFromKTable() {
+		StreamsBuilder builder = new StreamsBuilder();
+		@SuppressWarnings("unused")
+		final KTable<String,String> localTable = builder
+		.table("local",  
+			Consumed.with(Serdes.String(), Serdes.String()),
+			Materialized.<String,String,KeyValueStore<Bytes,byte[]>>as("localStore"))
+		;
+        final Properties config = new Properties();
+        config.put(StreamsConfig.APPLICATION_ID_CONFIG, "test-TopologyTestDriver-cleanup");
+        config.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "dummy:1234");
+        config.put(StreamsConfig.STATE_DIR_CONFIG, TestUtils.tempDirectory().getAbsolutePath());
+    	try (TopologyTestDriver testDriver = new TopologyTestDriver(builder.build(), config)) {
+    		final KeyValueStore<String,String> localStore = testDriver.getKeyValueStore("localStore");
+    		Assert.assertNotNull(localStore);
+    		Assert.assertNotNull(testDriver.getAllStateStores().get("localStore"));
+    		//
+        	final ConsumerRecordFactory<String,String> crf = new ConsumerRecordFactory<>(new StringSerializer(), new StringSerializer());
+    		testDriver.pipeInput(crf.create("local", "one", "TheOne"));
+    		//
+    		Assert.assertEquals("TheOne", localStore.get("one"));
+    	}
+    }
+
+    @Test
+    public void shouldFeedStoreViaProducerFromKTable() {
+		StreamsBuilder builder = new StreamsBuilder();
+		@SuppressWarnings("unused")
+		final KTable<String,String> localTable = builder
+		.table("local",  
+			Consumed.with(Serdes.String(), Serdes.String()),
+			Materialized.<String,String,KeyValueStore<Bytes,byte[]>>as("localStore"))
+		;
+        final Properties config = new Properties();
+        config.put(StreamsConfig.APPLICATION_ID_CONFIG, "test-TopologyTestDriver-cleanup");
+        config.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "dummy:1234");
+        config.put(StreamsConfig.STATE_DIR_CONFIG, TestUtils.tempDirectory().getAbsolutePath());
+    	try (TopologyTestDriver testDriver = new TopologyTestDriver(builder.build(), config)) {
+    		final KeyValueStore<String,String> localStore = testDriver.getKeyValueStore("localStore");
+    		Assert.assertNotNull(localStore);
+    		Assert.assertNotNull(testDriver.getAllStateStores().get("localStore"));
+    		//
+    	   	final ConsumerRecordFactory<String,String> crf = new ConsumerRecordFactory<>(new StringSerializer(), new StringSerializer());
+    		testDriver.pipeInput(crf.create("local", "one", "TheOne"));
+    		// just to serialize keys and values
+    		ConsumerRecord<byte[],byte[]> cr = crf.create("dummy", "two", "Second"); 
+    		testDriver.getProducer().send(new ProducerRecord<>("local", null, cr.timestamp(), cr.key(), cr.value()));
```

Comment:
What is the different of this to `testDriver.pipeInput()`? What value does it add?

Reply:
First of all I'ave an instance of Producer and second and more important in my case is that I need to inspect in the test what has been pushed in the topic.
Try to replace:
    		testDriver.getProducer().send(new ProducerRecord<>("global", null, cr.timestamp(), cr.key(), cr.value()));
with
    		testDriver.pipeInput(crf.create("global", "two", "Second"));
in shouldFeedStoreViaProducerFromGlobalKTable()
and you will see that the test fails.

Reply:
In advanceWallClockTime() the captureOutputRecords() is called which in turns feeds the outputRecordsByTopic map.